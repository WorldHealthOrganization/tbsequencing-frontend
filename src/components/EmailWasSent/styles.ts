import { css } from '@emotion/react';

export const styles = {
  h1: css({
    textAlign: 'center',
    '&:first-of-type': {
      marginTop: '20px',
    },
  }),
  imageWrapper: css({
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
  }),
  image: css({
    bottom: 0,
    position: 'absolute',
  }),
};
