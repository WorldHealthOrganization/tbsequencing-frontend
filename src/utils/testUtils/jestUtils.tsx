import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
