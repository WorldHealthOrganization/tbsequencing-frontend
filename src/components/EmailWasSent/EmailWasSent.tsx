/** @jsxImportSource @emotion/react */
import React from 'react';
import { styles } from './styles';
import H1 from '../typography/H1';
import imgPath from '../../assets/images/email-sent.png';

export const EmailWasSent = () => (
  <div>
    <H1 style={styles.h1}>Your request has been processed</H1>
    <H1 style={styles.h1}>
      If a user with this email exists, you will receive an email with further instructions
    </H1>
    <div css={styles.imageWrapper}>
      <img css={styles.image} src={imgPath} alt="email was sent" />
    </div>
  </div>
);

export default EmailWasSent;
