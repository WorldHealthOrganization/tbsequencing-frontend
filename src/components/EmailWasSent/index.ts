import EmailWasSent from './EmailWasSent';

export default EmailWasSent;
