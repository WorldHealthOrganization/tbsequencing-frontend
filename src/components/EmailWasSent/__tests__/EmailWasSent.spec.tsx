import React from 'react';
import { render, getTreeJSONWithRedux } from '../../../utils/testUtils/jestUtils';
import EmailWasSent from '../EmailWasSent';

describe('EmailWasSent component test', () => {
  test('renders EmailWasSent component without error', () => {
    render(
      <EmailWasSent />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <EmailWasSent />,
    );

    expect(tree).toMatchSnapshot();
  });
});
