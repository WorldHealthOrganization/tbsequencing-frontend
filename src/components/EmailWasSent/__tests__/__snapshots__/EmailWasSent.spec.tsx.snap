// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailWasSent component test Should render correctly 1`] = `
<div>
  <p
    className="MuiTypography-root MuiTypography-body1 css-fgh2mf-MuiTypography-root-BaseText"
  >
    Your request has been processed
  </p>
  <p
    className="MuiTypography-root MuiTypography-body1 css-fgh2mf-MuiTypography-root-BaseText"
  >
    If a user with this email exists, you will receive an email with further instructions
  </p>
  <div
    className="css-1oppeth-EmailWasSent"
  >
    <img
      alt="email was sent"
      className="css-18dv6dx-EmailWasSent"
      src="email-sent.png"
    />
  </div>
</div>
`;
